src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -36,6 +37,28 @@ public boolean contains(Person toCheck) {
         return internalList.stream().anyMatch(toCheck::isSamePerson);
     }
 
+    /**
+     * Returns the index of the Person with the same phone number.
+     *
+     * @param toCheck Phone number to check against the list
+     * @return index of the Person with the same phone number in the list.
+     */
+    public int findNum(Phone toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().map(Person::getPhone).collect(Collectors.toList()).indexOf(toCheck);
+    }
+
+    /**
+     * Returns the index of the Person with the same email.
+     *
+     * @param toCheck Email to check against the list
+     * @return index of the Person with the same email in the list
+     */
+    public int findEmail(Email toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().map(Person::getEmail).collect(Collectors.toList()).indexOf(toCheck);
+    }
+
```

Comment:
Yes, we will stick to using list for now because some of us tried hashmap but it did not work well with the entire structure. However, if anyone has a breakthrough with the use of hashmap, we can alter the implementation to fit hashmap since it is more efficient for each look-up.

Unfortunately, since it is using linear search on a list, the time complexity for delete is O(n).